File: app/src/main/java/com/stupidtree/hichat/ui/chat/YunmojiListAdapter.java
package com.stupidtree.hichat.ui.chat;


import android.content.Context;
import android.view.View;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.stupidtree.hichat.R;
import com.stupidtree.hichat.data.model.Yunmoji;
import com.stupidtree.hichat.ui.base.BaseListAdapter;
import com.stupidtree.hichat.ui.base.BaseViewHolder;

import java.util.List;

import butterknife.BindView;

/**
 * 表情列表的适配器
 */
class YunmojiListAdapter extends BaseListAdapter<Yunmoji, YunmojiListAdapter.YunmojiItemHolder> {


    public YunmojiListAdapter(Context mContext, List<Yunmoji> mBeans) {
        super(mContext, mBeans);
    }

    @Override
    protected int getLayoutId(int viewType) {
        return R.layout.activity_chat_yunmoji_item;
    }

    @Override
    public YunmojiItemHolder createViewHolder(View v, int viewType) {
        return new YunmojiItemHolder(v);
    }

    @Override
    protected void bindHolder(@NonNull YunmojiItemHolder holder, @Nullable Yunmoji data, int position) {
        Yunmoji yunmoji = mBeans.get(position);
        holder.image.setImageResource(yunmoji.getImageID());
        //表示当这项的图片点击时调用onItemClickListener
        if (mOnItemClickListener != null) {
            holder.image.setOnClickListener(view -> mOnItemClickListener.onItemClick(data, view, position));
        }
    }

    static class YunmojiItemHolder extends BaseViewHolder {
        @BindView(R.id.image)
        ImageView image;

        public YunmojiItemHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/ui/chat/YunmojiListAdapter.kt
package com.stupidtree.cloudliter.ui.chat

import android.content.Context
import android.view.View
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import com.stupidtree.cloudliter.data.model.Yunmoji
import com.stupidtree.cloudliter.databinding.ActivityChatYunmojiItemBinding
import com.stupidtree.cloudliter.ui.base.BaseListAdapter
import com.stupidtree.cloudliter.ui.base.BaseViewHolder
import com.stupidtree.cloudliter.ui.chat.YunmojiListAdapter.YunmojiItemHolder

/**
 * 表情列表的适配器
 */
internal class YunmojiListAdapter(mContext: Context?, mBeans: MutableList<Yunmoji>) : BaseListAdapter<Yunmoji, YunmojiItemHolder>(mContext!!, mBeans) {


    override fun bindHolder(holder: YunmojiItemHolder, data: Yunmoji?, position: Int) {
        val yunmoji = mBeans[position]
        holder.binding.image.setImageResource(yunmoji.imageID)
        //表示当这项的图片点击时调用onItemClickListener
        if (mOnItemClickListener != null) {
            holder.binding.image.setOnClickListener { view: View? -> data?.let { mOnItemClickListener!!.onItemClick(it, view, position) } }
        }
    }

    internal class YunmojiItemHolder(itemView: ActivityChatYunmojiItemBinding) : BaseViewHolder<ActivityChatYunmojiItemBinding>(itemView)

    override fun getViewBinding(parent: ViewGroup, viewType: Int): ViewBinding {
        return ActivityChatYunmojiItemBinding.inflate(mInflater,parent,false)
    }

    override fun createViewHolder(viewBinding: ViewBinding, viewType: Int): YunmojiItemHolder {
        return YunmojiItemHolder(viewBinding as ActivityChatYunmojiItemBinding)
    }
}